sdk/test-utils/recorder-new/MIGRATION.md

```diff
@@ -0,0 +1,160 @@
+});
+```
+
+When initialising your client in your test, you should pass in the recorder as follows:
+
+```ts
+const client = new MyServiceClient(/* ... insert options here ... */, { httpClient: recorder });
+```
+
+This will allow requests to be intercepted and redirected to the proxy tool.
+
+## Starting and stopping the recorder
+
+The recorder must now be started and stopped explicitly. At the beginning of your test, start the recorder as follows:
+
+```ts
+await recorder.start({
+  envSetupForPlayback: {
+    // Your environment variables (equivalent to the old recorder's replaceableVariables option). See the section on environment variables below for detail
+  }
+  // Other options, e.g. sanitizers (which replace the customizationsOnRecordings option)
+});
+```
+
+And at the end of your test, stop the recorder:
+
+```ts
+await recorder.stop();
+```
+
+It is important that `recorder.stop()` is called, or otherwise the next test will throw an error when trying to start the already started recorder. Additionally, it is important that both the `start` and `stop` calls are awaited, for similar reasons.
+
+## Environment variables
+
+In the legacy recorder, the `replaceableVariables` option could be used to specify environment variables that would be replaced in the recording and set during playback. This could be used to ensure that secrets and user-specific options do not appear in the recording body.
+
+The Unified Recorder client provides this functionality through the use of the `envSetupForPlayback` option, which is passed when `recorder.start` is called. Like the legacy recorder, it takes in an object mapping environment variables to what they should be replaced with in the recording. For example:
+
+```ts
+await recorder.start({
+  envSetupForPlayback: {
+    TABLES_SAS_CONNECTION_STRING: "fakeConnectionString"
+  }
+});
+```
+
+Under the hood, this is powered by the Unified Recorder's sanitizer functionality.
+
+**⚠️Important:** To access environment variables, you must use the `env` export made available from the **new** recorder. This ensures that environment variables are sourced from the correct location (using `process.env` and `dotenv` in Node, and using `window.__env__` via Mocha in the browser), and also means that the environment variables set in `envSetupForPlayback` are used in playback mode.
+
+## Customizations on recordings
+
+A powerful feature of the legacy recorder was its `customizationsOnRecordings` option, which allowed for arbitrary replacements to be made to recordings. The new recorder's analog to this is the sanitizer functionality.
+
+For a simple find/replace, a `GeneralRegexSanitizer` can be used. For example:
+
+```ts
+recorder.addSanitizers({
+  generalRegexSanitizers: [
+    regex: "find",
+    value: "replace"
+  ],
+});
+```
+
+This example would replace all instances of `find` in the recording with `replace`.
+
+Other sanitizers for more complex use cases are also available.
```

Comment:
Should we elaborate here? Or is there somewhere we can link to?

Reply:
We should write docs and elaborate examples for the whole API.
I'll start that.

Reply:
However, I would consider connection string and header removal to be other main common use cases, let's add them too in this document along with the generalRegexSanitizer?

Reply:
> We should write docs and elaborate examples for the whole API.

Let’s break that down into individual tasks so that we all can understand how far we are from where we want to go 😊